### PR TITLE
ts custom auth

### PR DESCRIPTION
**Pull request by johannes-vogel**
will be develivered with cds 7.6

### CODE REVIEW

**Review by sjvans — APPROVED** [review]

File: node.js/authentication.md [hunk 1]
```diff
@@ -404,6 +404,14 @@ module.exports = function custom_auth (req, res, next) {
 }
 ```
 
+The TypeScript equivalent has to use the default export.
+
+```ts
+export default function custom_auth(req: any, res: any, next: any) {
```
Comment (hakimio) [inline node.js/authentication.md:410]: You can use proper types:
```ts
import { Request } from '@sap/cds';

export default function custom_auth(request: Request, response: Response, next: Function) {}
```

Comment (johannes-vogel) [reply node.js/authentication.md:410]: The `request` here is the express request object. There might be types available for express as well but I'd prefer to have that snippet as simple as possible.

Comment (johannes-vogel) [reply node.js/authentication.md:410]: The `cds.Request` type is simply wrong as e. g. `query` has a completely different semantic in the express request object.

For proper type support one would need to enhance the request type of express. I'll double check for improving the snippet. If it grows too big, I'll add a comment in the snippet.

File: node.js/authentication.md [hunk 2]
```diff
@@ -404,6 +404,14 @@ module.exports = function custom_auth (req, res, next) {
...
+  // do your custom authentication ...
+}
+```
```
Comment (johannes-vogel) [inline node.js/authentication.md:413]:
```suggestion
```ts
import cds from "@sap/cds";
import {Request, Response, NextFunction} from "express";
type Req = Request & { user: cds.User, tenant: string };

export default function custom_auth(req: Req, res: Response, next: NextFunction) {
  // do your custom authentication ...
}
```
```

Comment (hakimio) [reply node.js/authentication.md:413]: Looks good to me.

Comment (hakimio) [reply node.js/authentication.md:413]: Just a side note: maybe this type should be included in `cds-types`:
```ts
type AuthRequest = Request & { user: cds.User; tenant: string };
```

File: node.js/authentication.md [hunk 3]
```diff
@@ -404,6 +404,17 @@ module.exports = function custom_auth (req, res, next) {
...
+
+```ts
+import cds from "@sap/cds";
+import {Request, Response, NextFunction} from "express";
+type Req = Request & { user: cds.User, tenant: string };
+
+export default function custom_auth(req: Req, res: Response, next: NextFunction) {
+  // do your custom authentication ...
+}
+
```
Comment (renejeglinsky) [inline node.js/authentication.md:417]:
```suggestion
```
```

**Review by renejeglinsky — APPROVED** [review]

**Pull request merged**